### ISSUE TITLE
Mapi Listener does not like the "obfuscate" argument

### ISSUE THREAD

**Issue by sharpbazil** [cobbr/ObfuscatedEmpire#14]
Empire 2.0, Ubuntu 16.04
Always found the MAPI listener/stager to be finnicky, and it doesn't seem to like most things about being included with empire. This particular issue appears unique to ObfuscatedEmpire. (I tend to have to manually enter the variables in the script.
The error is on generation of any powershell stager for the MAPI listener:
[!] Exception: generate_launcher() got an unexpected keyword argument 'obfuscate'
Might just be a quick fix to delete wherever that argument is sent to the MAPI stager generation code.

**Comment by cjensenius** [comment]
This issue exists in the http_foreign listener when using obfuscation in generate_launcher, I removed the extra argument and this resolved the issue.

**Comment by cobbr** [comment]
@sharpbazil Thanks for the report, sorry I haven't had a change to take a look. I'll take a look when I get a chance.

@cjensenius If you have a fix for this, could you submit a PR so I can understand how you fixed it?

**Comment by cjensenius** [comment]
@cobbr, no problem I will send a PR sometime tomorrow after testing, though in my haste earlier I neglected to notice that @sharpbazil has a simple function signature issue which is distinct from the issue I ran into. Would you prefer a PR for each issue separately?

OP issue (I think):
The mapi listener is missing the obfuscate named argument but it is receiving it (line 154)
`def generate_launcher(self, encode=True, userAgent='default', proxy='default', proxyCreds='default', stagerRetries='0', language=None, safeChecks='', listenerName=None):`

My change to the foreign listener was to line 232 of https://github.com/cobbr/ObfuscatedEmpire/blob/master/lib/listeners/http_foreign.py

`stager = helpers.obfuscate(stager, self.mainMenu.installPath, obfuscationCommand=obfuscationCommand)`
to
`stager = helpers.obfuscate(stager, obfuscationCommand=obfuscationCommand)`

**Comment by cobbr** [comment]
@cjensenius Yeah, if you have identified two distinct issues, then it would probably be helpful to have a PR for each of them. Either way works though.
